destination/destination_integration_test.go

```diff
@@ -0,0 +1,349 @@
+
+		err := d.Configure(ctx, map[string]string{
+			config.ConfigAddress:       "localhost:2222",
+			config.ConfigHostKey:       hostKey,
+			config.ConfigUsername:      "user",
+			config.ConfigPassword:      "pass",
+			config.ConfigDirectoryPath: "/upload",
+		})
+		is.NoErr(err)
+
+		err = d.Open(ctx)
+		is.NoErr(err)
+
+		content := []byte(`Hello World!`)
+
+		records := []opencdc.Record{
+			sdk.Util.Source.NewRecordCreate(
+				nil,
+				map[string]string{
+					opencdc.MetadataCollection: "upload",
+					opencdc.MetadataCreatedAt:  time.Now().UTC().Format(time.RFC3339),
+					"filename":                 "example.txt",
+					"source_path":              "/upload",
+					"file_size":                fmt.Sprintf("%d", len(content)),
+					"mod_time":                 time.Now().UTC().Format(time.RFC3339),
+				},
+				opencdc.StructuredData{"filename": "example.txt"},
+				opencdc.RawData(content),
+			),
+		}
+
+		n, err := d.Write(ctx, records)
+		is.NoErr(err)
+		is.Equal(n, len(records))
+
+		d.Teardown(ctx)
+	})
+
+	t.Run("destination write success filename from key", func(t *testing.T) {
+		is := is.New(t)
+		d := NewDestination()
+		ctx := context.Background()
+
+		err := d.Configure(ctx, map[string]string{
+			config.ConfigAddress:       "localhost:2222",
+			config.ConfigHostKey:       hostKey,
+			config.ConfigUsername:      "user",
+			config.ConfigPassword:      "pass",
+			config.ConfigDirectoryPath: "/upload",
+		})
+		is.NoErr(err)
+
+		err = d.Open(ctx)
+		is.NoErr(err)
+
+		content := []byte(`Hello World!`)
+
+		records := []opencdc.Record{
+			sdk.Util.Source.NewRecordCreate(
+				nil,
+				map[string]string{
+					opencdc.MetadataCollection: "upload",
+					opencdc.MetadataCreatedAt:  time.Now().UTC().Format(time.RFC3339),
+					"source_path":              "/upload",
+					"file_size":                fmt.Sprintf("%d", len(content)),
+					"mod_time":                 time.Now().UTC().Format(time.RFC3339),
+				},
+				opencdc.StructuredData{"filename": "example.txt"},
+				opencdc.RawData(content),
+			),
+		}
+
+		n, err := d.Write(ctx, records)
+		is.NoErr(err)
+		is.Equal(n, len(records))
+
+		d.Teardown(ctx)
+	})
+	t.Run("destination write failure", func(t *testing.T) {
+		is := is.New(t)
+		d := NewDestination()
+		ctx := context.Background()
+
+		err := d.Configure(ctx, map[string]string{
+			config.ConfigAddress:       "localhost:2222",
+			config.ConfigHostKey:       hostKey,
+			config.ConfigUsername:      "user",
+			config.ConfigPassword:      "pass",
+			config.ConfigDirectoryPath: "/upload",
+		})
+		is.NoErr(err)
+
+		err = d.Open(ctx)
+		is.NoErr(err)
+
+		content := []byte(`Hello World!`)
+
+		records := []opencdc.Record{
+			sdk.Util.Source.NewRecordCreate(
+				nil,
+				map[string]string{
+					opencdc.MetadataCollection: "upload",
+					opencdc.MetadataCreatedAt:  time.Now().UTC().Format(time.RFC3339),
+					"filename":                 "",
+					"source_path":              "/upload",
```

Comment:
Hmm, what's this used for?

Reply:
This is the case where filename is empty and it will give error while creating the file.

Reply:
Gotcha... How about source_path?

Reply:
"source_path" is metadata added in the source. Not really useful in destination.